pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -152,15 +152,6 @@ type PipelineDeclaredResource struct {
 	Type PipelineResourceType `json:"type"`
 }
 
-// PipelineConditionResource allows a Pipeline to declare how its DeclaredPipelineResources
-// should be provided to a Condition as its inputs.
-type PipelineConditionResource struct {
```

Comment:
As far as I understand this was removed to make the logic in the code simpler, since you have to deal with `PipelineTaskInputResource` only; I wonder if we may need different behaviours for task input resources and condition resources, but we can still add this back if becomes necessary.

Reply:
Yeah, I don't think we do at the moment; but if we do we can add this back later on. Right now, this is basically an unused/extra level of indirection